### ISSUE TITLE
Vim automatically downgrades after some time

### ISSUE THREAD

**Issue by Tiriel** [vim/vim#2919]
Hi everyone,

Sorry to bother you, but I have to admit I'm completely puzzled. I've recently started using Vim everyday for development, both personal and professional. Needing a good list of plugins and extensions, I decided to compile it. After reading a lot of doc, removing every Vim version currently on my computer, and resolving dependency issues, I got it running pretty well, no problem, everything running smoothly. `vim --version` displayed correctly version 8.0 with all the features I wanted.

And then, the following day, when starting Vim, I had an error telling me that the termguicolors option didn't exist. Quickly verifying, `vim --version` displayed version 7.4.

To be honest, I have no idea why this occurred, and it continues to occur quite regularly. Every two days, I have to remove every vim version on my computer and recompile it.

The set up:
* OS: Ubuntu 16.04 and 18.04
* installation method: manual compiling, with `checkinstall` in place of `make install` (also occurred with `make install`)
* Configure options:
```
./configure \
--enable-luainterp \
--enable-perlinterp \
--enable-pythoninterp \
--enable-rubyinterp \
--enable-cscope \
--disable-netbeans \
--enable-multibyte \
--enable-largefile \
--enable-gui=no \
--with-features=huge \
--with-ruby-command=/usr/bin/ruby \
--with-python-config-dir=/usr/lib/python2.7/config-x86_64-linux-gnu \
--enable-python3interp \
--with-python3-config-dir=/usr/lib/python3.5/config-3.5m-x86_64-linux-gnu \
--with-luajit \
--with-x \
--enable-fontset \
--enable-largefile \
--with-compiledby="Benjamin Zaslavsky" \
--enable-fail-if-missing
```
* Good `vim --version` output:
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled May 16 2018 09:58:58)
Rustines incluses : 1-1848
Compilé par Benjamin Zaslavsky
Énorme version sans interface graphique.
  Fonctionnalités incluses (+) ou non (-) :
+acl               +farsi             +mouse_sgr         -tag_any_white
+arabic            +file_in_path      -mouse_sysmouse    -tcl
+autocmd           +find_in_path      +mouse_urxvt       +termguicolors
-autoservername    +float             +mouse_xterm       +terminal
-balloon_eval      +folding           +multi_byte        +terminfo
+balloon_eval_term -footer            +multi_lang        +termresponse
-browse            +fork()            -mzscheme          +textobjects
++builtin_terms    +gettext           -netbeans_intg     +timers
+byte_offset       -hangul_input      +num64             +title
+channel           +iconv             +packages          -toolbar
+cindent           +insert_expand     +path_extra        +user_commands
+clientserver      +job               +perl              +vertsplit
+clipboard         +jumplist          +persistent_undo   +virtualedit
+cmdline_compl     +keymap            +postscript        +visual
+cmdline_hist      +lambda            +printer           +visualextra
+cmdline_info      +langmap           +profile           +viminfo
+comments          +libcall           +python/dyn        +vreplace
+conceal           +linebreak         +python3/dyn       +wildignore
+cryptv            +lispindent        +quickfix          +wildmenu
+cscope            +listcmds          +reltime           +windows
+cursorbind        +localmap          +rightleft         +writebackup
+cursorshape       +lua               +ruby              +X11
+dialog_con        +menu              +scrollbind        +xfontset
+diff              +mksession         +signs             -xim
+digraphs          +modify_fname      +smartindent       -xpm
-dnd               +mouse             +startuptime       +xsmp_interact
-ebcdic            -mouseshape        +statusline        +xterm_clipboard
+emacs_tags        +mouse_dec         -sun_workshop      -xterm_save
+eval              -mouse_gpm         +syntax            
+ex_extra          -mouse_jsbterm     +tag_binary        
+extra_search      +mouse_netterm     +tag_old_static    
         fichier vimrc système : "$VIM/vimrc"
     fichier vimrc utilisateur : "$HOME/.vimrc"
 2me fichier vimrc utilisateur : "~/.vim/vimrc"
      fichier exrc utilisateur : "$HOME/.exrc"
 fichier de valeurs par défaut : "$VIMRUNTIME/defaults.vim"
               $VIM par défaut : "/usr/local/share/vim"
Compilation : gcc -c -I. -Iproto -DHAVE_CONFIG_H     -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Édition de liens : gcc   -L. -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,-E   -L/usr/local/lib -Wl,--as-needed -o vim    -lSM -lICE -lXpm -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo -lnsl  -lselinux   -ldl  -L/usr/lib/x86_64-linux-gnu -lluajit-5.1 -Wl,-E  -fstack-protector-strong -L/usr/local/lib  -L/usr/lib/x86_64-linux-gnu/perl/5.22/CORE -lperl -ldl -lm -lpthread -lcrypt    -lruby-2.3 -lpthread -lgmp -ldl -lcrypt -lm  
```
* 'Bad' `vim --version` output:
```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:44:48)
Rustines incluses : 1-1689
Rustines extra : 8.0.0056
Modifié par pkg-vim-maintainers@lists.alioth.debian.org
Compilé par pkg-vim-maintainers@lists.alioth.debian.org
Énorme version sans interface graphique.
  Fonctionnalités incluses (+) ou non (-) :
+acl             +farsi           +mouse_netterm   +tag_binary
+arabic          +file_in_path    +mouse_sgr       +tag_old_static
+autocmd         +find_in_path    -mouse_sysmouse  -tag_any_white
-balloon_eval    +float           +mouse_urxvt     -tcl
-browse          +folding         +mouse_xterm     +terminfo
++builtin_terms  -footer          +multi_byte      +termresponse
+byte_offset     +fork()          +multi_lang      +textobjects
+channel         +gettext         -mzscheme        +timers
+cindent         -hangul_input    +netbeans_intg   +title
-clientserver    +iconv           +packages        -toolbar
-clipboard       +insert_expand   +path_extra      +user_commands
+cmdline_compl   +job             -perl            +vertsplit
+cmdline_hist    +jumplist        +persistent_undo +virtualedit
+cmdline_info    +keymap          +postscript      +visual
+comments        +langmap         +printer         +visualextra
+conceal         +libcall         +profile         +viminfo
+cryptv          +linebreak       -python          +vreplace
+cscope          +lispindent      +python3         +wildignore
+cursorbind      +listcmds        +quickfix        +wildmenu
+cursorshape     +localmap        +reltime         +windows
+dialog_con      -lua             +rightleft       +writebackup
+diff            +menu            -ruby            -X11
+digraphs        +mksession       +scrollbind      -xfontset
-dnd             +modify_fname    +signs           -xim
-ebcdic          +mouse           +smartindent     -xsmp
+emacs_tags      -mouseshape      +startuptime     -xterm_clipboard
+eval            +mouse_dec       +statusline      -xterm_save
+ex_extra        +mouse_gpm       -sun_workshop    -xpm
+extra_search    -mouse_jsbterm   +syntax          
         fichier vimrc système : "$VIM/vimrc"
     fichier vimrc utilisateur : "$HOME/.vimrc"
 2me fichier vimrc utilisateur : "~/.vim/vimrc"
      fichier exrc utilisateur : "$HOME/.exrc"
               $VIM par défaut : "/usr/share/vim"
Compilation : gcc -c -I. -Iproto -DHAVE_CONFIG_H   -Wdate-time  -g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1      
Édition de liens : gcc   -Wl,-Bsymbolic-functions -fPIE -pie -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -o vim        -lm -ltinfo -lnsl  -lselinux  -lacl -lattr -lgpm -ldl     -L/usr/lib/python3.5/config-3.5m-x86_64-linux-gnu -lpython3.5m -lpthread -ldl -lutil -lm 
```

If anyone has an idea why this problem occures, it would greatly help me. I searched the interwebs and coundl'nt find anything yet.

Thanks!

**Comment by inkarkat** [comment]
It is possible to keep the default Vim version (that is installed by your Linux distribution) _and_ have a (newer) custom-built Vim in parallel. Ubuntu's package management installs (Vim and other packages) under `/usr/bin`, whereas self-compiled binaries go into `/usr/local/bin` by convention (and this is the location that Vim's configure script defaults to). Users usually select their preferred version by adapting the order or augmenting the user's `PATH` variable: If you put `/usr/local/bin` before `/usr/bin`, the self-compiled version should be launched with `vim`. (You can always explicitly launch the fallback via `/usr/bin/vim`, too.)

You seem to have `/usr/bin` before `/usr/local/bin`, and your package manager may regularly reinstall Vim on its own (e.g. when applying security updates), because it thinks the Vim package is missing (that can happen if you forcibly uninstall a package even though there are dependencies to it, or you've circumvented the package manager somehow).

TL;DR: Look at your `PATH` definition (likely in `/etc/profile`, or `~/.bashrc`), and put `/usr/local/bin` up front.

---

If you still have questions about this, please don't use Vim's issue tracker; it's meant for core Vim issues, user questions and environment issues are off-topic here. (This issue will probably be closed soon.) There's a separate [vim mailing list for users](http://www.vim.org/community.php); or just ask at the [Ask Ubuntu](https://askubuntu.com/) or [Vim StackExchange](https://vi.stackexchange.com/) sites.

**Comment by Tiriel** [comment]
Awesome, thanks for your quick answer, and sorry for the bother. I quite didn't know where to post this. Going to close it on my own.

Thanks again!
